### PR TITLE
#24203 prevent negativ points in StudyProgramme

### PR DESCRIPTION
**Pull request by shecken**
https://mantis.ilias.de/view.php?id=24203

### CODE REVIEW

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php [hunk 1]
```diff
@@ -166,21 +168,22 @@ protected function update() {
...
-			ilUtil::sendSuccess($this->lng->txt("msg_obj_modified"),true);
-			$response = ilAsyncOutputHandler::encodeAsyncResponse(array("success"=>true, "message"=>$this->lng->txt("msg_obj_modified")));
-		} else {
-			ilUtil::sendFailure($this->lng->txt("msg_form_save_error"));
-			$response = ilAsyncOutputHandler::encodeAsyncResponse(array("success"=>false, "errors"=>$form->getErrors()));
-		}
-
 		if($this->ctrl->isAsynch()) {
+			if ($update_possible) {
+				$this->updateWith($prg, $content);
```
Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php]: Dann bitte noch eine separate Mail an Alexandra mit Martin/Stefan/mir im CC

Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php]: This needs to be executed in the non-asynch-case as well. I would suggest to leave it where it is.

Comment (shecken) [reply Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php]: @klees I changed this.

**Review by klees — CHANGES_REQUESTED** [review]
Hi @shecken, thx for working on this. There still seem to be problems here. As far as I understand your changes, we will never return anything to the client in the async-case because the line `return ilAsyncOutputHandler::handleAsyncOutput($form->getHTML(), $response, false);` was removed.

**Pull request merged**